src/Pages/Signup/Signup.js

```diff
@@ -1,12 +1,480 @@
 import React, { Component } from "react";
+import "./Signup.scss";
 
 class Signup extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      isModalActive: false,
+      showIdDetail: false,
+      showPwDetail: false,
+      showCheckPwDetail: false,
+      validateId: false,
+      validatePw: false,
+      validatePwLength: false,
+      isPwSame: false,
+
+      account: "",
+      password: "",
+      name: "",
+      email: "",
+      phoneNumber: "",
+      gender: "",
+      recommend: "",
+      year: "",
+      month: "",
+      day: "",
+      agree: [false, false, false, false, false, false, false],
+    };
   }
+
+  getValue = e => {
+    const { id, value } = e.target;
+    this.setState({ [id]: value });
+  };
+
+  handleModal = () => {
+    this.setState({ isModalActive: !this.state.isModalActive });
+  };
+
+  openIdDetail = () => {
+    this.setState({ showIdDetail: true });
+  };
+
+  openPwDetail = () => {
+    this.setState({ showPwDetail: true });
+  };
+
+  openCheckPwDetail = () => {
+    this.setState({ showCheckPwDetail: true });
+  };
+
+  validationId = e => {
+    const { value } = e.target;
+    const txt = /^[a-z]{6,}$/;
+    const numtxt6 = /^(?=.*[0-9])(?=.*[a-zA-Z]).{6,}$/;
+    const idValidation = txt.test(value) || numtxt6.test(value);
+    this.setState({ account: value });
+    this.setState({ validateId: idValidation ? true : false });
```

Comment:
setState 하나에 다 넣어도 될 것 같네요!
그리고 boolean 값이라서 true, false를 직접 명시하지 않고도 바로 사용할 수 있을 것 같습니다!
```suggestion
    this.setState({
      account: value,
      validateId: idValidation,
    });
```

Reply:
`validateId: idValidation`  이렇게 할수  있는지 몰랐습니다. 신기하네요 ㅎㅎㅎㅎ